index.js

```diff
@@ -103,22 +91,23 @@ WuiDom.prototype._assign = function (tagName, options) {
 		}
 	} else if (tagName instanceof window.Element) {
 		// the first passed argument already is a real HTML Element
-
 		this.rootElement = tagName;
 	} else {
 		throw new Error('WuiDom.assign requires the given argument to be a DOM Element or tagName.');
 	}
 
-	if (options && options.name) {
-		this._name = options.name;
-	}
+	options = options || {};
 
-	if (options && options.hidden) {
-		// start hidden
+	// start hidden
+	if (options.hidden) {
 		this.hide();
 	}
 
-	return this.rootElement;
+	// set identifier (used by getChild)
+	this._name = String(options.name);
+
+	this.addClassNames(options.className);
+	this.setStyles(options.style || {});
```

Comment:
Why call functions that are going to do absolutely nothing? Why all this deoptimization?